### DISCUSSION TITLE
Only using W 2022 server

### DISCUSSION THREAD

**Discussion by Bazastro** [akeeba/panopticon#251] [General]
I do have a server farm, and several subnets. My running Joomla sites on couple W 2022 servers are:
PHP Built On	Windows NT XXX 10.0 build 20348 (Windows Server 2022) AMD64
Database Type	mysql
Database Version	8.0.32
Database Collation	utf8mb4_unicode_ci
Database Connection Collation	utf8mb4_0900_ai_ci
Database Connection Encryption	None
Database Server Supports Connection Encryption	Yes
PHP Version	8.2.3
Web Server	Microsoft-IIS/10.0
WebServer to PHP Interface	cgi-fcgi
Joomla! Version	Joomla! 5.0.0 Stable [ Kuboresha ] 17-October-2023 16:00 GMT
Joomla Backward Compatibility Plugin	Enabled ()
User Agent	Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36 Edg/119.0.0.0
I'm warming up to the idea of site with a hundred eyes with which I can savage all my monitored sites.
But, I reckon it may need a site for itself outside of my IP subnets. I've only been thinking it through for an hour so far.
Sites are SSL only, from within the subnets, it makes connections difficult.
Your thoughts?
Cheers, Baz.

**Comment by nikosdion** [comment]
I am really not a Windows person —I love my Linux and macOS— but I don't see any reason why you would have a problem. You have a fairly recent version of PHP which could use a bit of an update (8.2.12 is the current release and it has security fixes over your installed version), and your database version is definitely the preferred and supported one.

You will definitely need a lot of Windows Scheduled Tasks because Windows in general, as an Operating System, is _really slow_ when it comes to filesystem access. It all comes down to how Microsoft decided to cripple filesystem caching back in the NT days, but that's another discussion for another time.

Regarding site access, I don't know what is your network setup exactly but I believe that if you have hundreds of sites they are Internet-accessible. In this case, you'd just use their public domain name when configuring them with Panopticon. I've done something similar for my test installation, albeit on Linux.

**Reply by Bazastro** [reply]
I guess you get a big shout out for still porting your products in windoze. I'm going to come back in a few days to revisit the site maintenance concept -it's a good one. My web servers are on different private subnets, access from within my own farm unlikely with a jump box as more than one DNS server, possible running it offsite, as you say. After I've replaced all my MK routers & schema. Must do much other user related work yet in other areas.
Thanks for the heads up re PHP updating - yeah I saw it, big job for me, has to be done.
Cheers.

**Reply by EngNashar** [reply]
The problem is that IIS as a windows web server dose not deal with language like PHP with no compiler and PHP cgi retired since last version on 2022 so if you change your server to apache for example I think the problem will be solved  but i think at this moment you will not be able to launch even a test without buying a domain

**Reply by nikosdion** [reply]
PHP CGI/FastCGI is doing great, thank you.

What is no longer recommended is using the Apache module and this has been the case for nearly fifteen years. 

Here is what I do. https://www.dionysopoulos.me/iis-development-server-for-php-on-windows-10.html I have this on Windows 11 as well, with PHP 8.2.

**Reply by Bazastro** [reply]
Agree, Nick. The "rename the web.config.txt file to web.config to enable SEF URLs" is one of things I'm struggling with, I'm reluctant to fiddle with running sites unless I know outcomes will meet expectations. I've no DNS worries, being unlucky enough to have a server farm.
Good article, impressive touch typing skills.
I'm considering running a PHP version in separate folders for each site, as running websites with each requiring different PHP requirements & features - opens doors a little wide. In windoze, just upgrading PHP versions is enough of a pita for one, only happens once or twice a year, I'm thinking just do it. BeyondCompare is my friend re ini files. Thanks & cheers.

**Reply by nikosdion** [reply]
The instruction is straightforward and the web.config.txt in Joomla is **very** basic. It won't upset any running sites.

Changing PHP versions in Windows is super easy. Make sure you have the right Visual C++ Runtime installed, unzip the new version, copy over the configuration, create a new version with Microsoft's PHP version manager for IIS, and you can switch over sites with a downtime of approximately 1 second. You can go as old as PHP 5.3 and as new as PHP 8.3 **on the same box**. This is not an option we have in Linux or macOS unless we use Docker containers to host the PHP-FPM server (which is a PITA on its own right).

Windows is very easy, but it's also dog slow unless you use Windows Server. Even Windows Professional prioritizes application memory over disk caching which makes for a horrid throughput even on a blazing fast PCIe 4 NVMe. Between that and the _laissez faire_ attitude on filesystem case I could not use Windows to develop software which is why I moved to UNIX Operating Systems (Linux and macOS, the latter essentially being BSD with a custom Mach kernel and a proprietary desktop environment).
